# Exploring eBay Car Sales Data

I investigated the relationship between price and brand of cars sold in eBay Kleinanzeigen, a classifieds section of the German eBay website.

I choose top 6 brands, namely Volkswagen, BMW, Opel, Mercedes, Audi and Ford in terms of the number of cars sold.

The findings are following:

| Brand |Mean Price |Mean Mileage |
| :--- | :---: | ---: |
| Volkswagen | 5604 | 128711 |
| BWW | 8333 | 132573 |
| Opel | 2975 | 129310 |
| Mercedes | 8333 | 132573 |
| Audi | 9337 | 129157 |
| Ford | 4055 | 124266 |

There is not siginigicant differece abount mean milage between each brand.  
Therefore, the differece about price can be regarded as the value of brand.

*The dataset was originally scraped and uploaded to Kaggle and Data Quest have made a few modifications from the original dataset that was uploaded to Kaggle:

In [50]:
import pandas as pd
import numpy as np

In [51]:
autos = pd.read_csv("autos.csv", encoding="Latin-1")

In [52]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [53]:
autos.head(5)

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Vehicle type, gearbox,model, fuel type,not repaired damage has null value.
Most of columns values are object.
These two points should be cleaned.

In [54]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [55]:
autos.rename(columns={'yearOfRegistration':'registration_year',
                     'monthOfRegistration':'registration_month',
                     'notRepairedDamage':'unrepaired_damage',
                     'dateCreated':'ad_created',
                      'dateCrawled':'date_crawled',
                     'offerType':'offer_type',
                     'vehicleType':'vehicle_type',
                     'powerPS':'power_ps',
                     'fuelType':'fuel_type',
                     'nrOfPictures':'num_pictures',
                     'postalCode':'postal_code',
                     'lastSeen':'last_seen'},inplace=True)

In [56]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  num_pictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

For the improvement of readability of dataframe, names of colums are modified following rules:
Camel case > Snake case
OF is omitted
not repaired > unrepaired
dateCreated > ad_created

In [57]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-23 18:39:34  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  num_pictures   postal_code            last_seen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

num_prictures only has the value of 0 and should be dropped.  
registration year's max and min are not realistic value, which means that I need more investigation.  
min of power_ps and registration month should ne more thatn 0.  
max of power_ps is too high.  
price and otometer should be numeric.  

In [58]:
autos['price'].unique()

array(['$5,000', '$8,500', '$8,990', ..., '$385', '$22,200', '$16,995'],
      dtype=object)

In [59]:
def clean_price(d):
    d = d.str.replace('$','')
    d = d.str.replace(',','')
    d = d.astype(int)
    return d

In [60]:
autos['price'] = clean_price(autos['price'])

In [61]:
autos['odometer'].unique()

array(['150,000km', '70,000km', '50,000km', '80,000km', '10,000km',
       '30,000km', '125,000km', '90,000km', '20,000km', '60,000km',
       '5,000km', '100,000km', '40,000km'], dtype=object)

In [62]:
def clean_odometer(d):
    d = d.str.replace('km','')
    d = d.str.replace(',','')
    d = d.astype(int)
    return d

In [63]:
autos['odometer'] = clean_odometer(autos['odometer'])

In [64]:
autos.rename(columns={"odometer":"odometer_km"},inplace=True)

In [65]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type  price   abtest vehicle_type  registration_year  \
0  privat    Angebot   5000  control          bus               2004   
1  privat    Angebot   8500  control    limousine               1997   
2  privat    Angebot   8990     test    limousine               2009   
3  privat    Angebot   4350  control   kleinwagen               2007   
4  privat    Angebot   1350     test        kombi               2003   

     gearbox  power_ps   model  odometer_km  registration_month fuel_type  \
0    manuell       158  andere       150000                   3       lpg   
1  automatik       286     7er       150000                   6    benzin   
2    manuell       102    golf        70000                   7    benzin   
3  automatik        71  fortwo        70000                   6    benzin   
4    manuell         0   focus       150000                   7    benzin   

        brand unrepaired_damage           ad_created  num_pictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

In [66]:
autos['price'].unique().shape

(2357,)

In [67]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [68]:
autos['price'][autos['price']<100].value_counts()

0     1421
1      156
50      49
99      19
80      15
70      10
60       9
30       7
10       7
40       6
75       5
90       5
25       5
65       5
49       4
45       4
20       4
2        3
17       3
12       3
55       2
5        2
15       2
13       2
11       2
9        1
89       1
3        1
8        1
14       1
66       1
47       1
18       1
29       1
59       1
35       1
79       1
Name: price, dtype: int64

In [69]:
autos['price'][autos['price']>1000000].value_counts()

12345678    3
11111111    2
99999999    1
3890000     1
27322222    1
1234566     1
10000000    1
1300000     1
Name: price, dtype: int64

In [70]:
autos[autos['price'].between(1,1000000)]

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offer_type  price   abtest vehicle_type  registration_year  \
0      privat    Angebot   5000  control          bus               2004   
1      privat    Angebot   8500  control    limousine               1997   
2      privat    Angebot   8990     test    limousine               2009   
3      privat    Angebot   4350  control   kleinwagen               2007   
4      privat    Angebot   1350     test        kombi               2003   
...       ...        ...    ...      ...          ...                ...   
49995  privat    Angebot  24900  control    limousine               2011   
49996  privat    Angebot   1980  control       cabrio               1996   
49997  privat    Angebot  13200     test       cabrio               2014   
49998  privat    Angebot  22900  control        kombi               2013   
49999  privat    Angebot   1250  control    limousine               1996   

         gearbox  power_ps   model  odometer_km  registration_month fuel_type  \
0        manuell       158  andere       150000                   3       lpg   
1      automatik       286     7er       150000                   6    benzin   
2        manuell       102    golf        70000                   7    benzin   
3      automatik        71  fortwo        70000                   6    benzin   
4        manuell         0   focus       150000                   7    benzin   
...          ...       ...     ...          ...                 ...       ...   
49995  automatik       239      q5       100000                   1    diesel   
49996    manuell        75   astra       150000                   5    benzin   
49997  automatik        69     500         5000                  11    benzin   
49998    manuell       150      a3        40000                  11    diesel   
49999    manuell       101  vectra       150000                   1    benzin   

            brand unrepaired_damage           ad_created  num_pictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postal_code            last_seen  
0            79588  2016-04-06 06:45:54  
1            71034  2016-04-06 14:45:08  
2            35394  2016-04-06 20:15:37  
3            33729  2016-03-15 03:16:28  
4            39218  2016-04-01 14:38:50  
...           

For the price column, the value less than 1 and larger than 1,000,000 are reagarded as outliers and dropped.  
0 is not realistic for Ebey.  
Prices over 1M looks strange (e.g. 123456, 99999999) and too big.

In [74]:
autos['odometer_km'].unique().shape

(13,)

In [75]:
autos['odometer_km'].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [76]:
autos['odometer_km'].value_counts()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64

For odometer_km, there are not so many unique value and all of them are realistic.  
So, I do not remove any value.


In [77]:
autos = autos[autos['price'].between(1,1000000)]

In [83]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer_km', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [92]:
autos['date_crawled'] = pd.to_datetime(autos['date_crawled'].str[:10], format = "%Y-%m-%d")
autos['ad_created'] = pd.to_datetime(autos['ad_created'].str[:10], format =  "%Y-%m-%d")
autos['last_seen'] = pd.to_datetime(autos['last_seen'].str[:10], format = "%Y-%m-%d")

In [103]:
print('date_crawled:\n', autos['date_crawled'].value_counts(normalize=True, dropna=False).sort_index())
print('ad_created:\n', autos['ad_created'].value_counts(normalize=True, dropna=False).sort_index())
print('last_seen:\n', autos['last_seen'].value_counts(normalize=True, dropna=False).sort_index())

date_crawled:
 2016-03-05    0.025325
2016-03-06    0.014042
2016-03-07    0.036011
2016-03-08    0.033294
2016-03-09    0.033088
2016-03-10    0.032182
2016-03-11    0.032573
2016-03-12    0.036917
2016-03-13    0.015669
2016-03-14    0.036547
2016-03-15    0.034282
2016-03-16    0.029608
2016-03-17    0.031646
2016-03-18    0.012910
2016-03-19    0.034776
2016-03-20    0.037885
2016-03-21    0.037391
2016-03-22    0.032985
2016-03-23    0.032223
2016-03-24    0.029340
2016-03-25    0.031605
2016-03-26    0.032202
2016-03-27    0.031090
2016-03-28    0.034858
2016-03-29    0.034117
2016-03-30    0.033685
2016-03-31    0.031832
2016-04-01    0.033685
2016-04-02    0.035476
2016-04-03    0.038606
2016-04-04    0.036485
2016-04-05    0.013095
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64
ad_created:
 2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
                ...   
2016-04-03    0.0

In [106]:
autos['ad_created'].describe()

count                   48568
unique                     76
top       2016-04-03 00:00:00
freq                     1887
first     2015-06-11 00:00:00
last      2016-04-07 00:00:00
Name: ad_created, dtype: object

date_crawled and last_seen has same extent of date.
ad_created 

In [107]:
autos['registration_year'].describe()

count    48568.000000
mean      2004.754612
std         88.641262
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

min and max are unrealistic

In [109]:
autos['registration_year'].value_counts().sort_index().head(20)

1000    1
1001    1
1111    1
1800    2
1910    5
1927    1
1929    1
1931    1
1934    2
1937    4
1938    1
1939    1
1941    2
1943    1
1948    1
1950    3
1951    2
1952    1
1953    1
1954    2
Name: registration_year, dtype: int64

I decide that the year of 1910-2016 is acceptable.

In [113]:
autos = autos[autos['registration_year'].between(1910,2016)]

In [119]:
autos['registration_year'].value_counts(normalize=True).tail(30)

1966    0.000471
1976    0.000450
1969    0.000407
1975    0.000386
1965    0.000364
1964    0.000257
1963    0.000171
1959    0.000129
1961    0.000129
1910    0.000107
1956    0.000086
1958    0.000086
1937    0.000086
1962    0.000086
1950    0.000064
1954    0.000043
1941    0.000043
1951    0.000043
1934    0.000043
1957    0.000043
1955    0.000043
1953    0.000021
1943    0.000021
1929    0.000021
1939    0.000021
1938    0.000021
1948    0.000021
1927    0.000021
1931    0.000021
1952    0.000021
Name: registration_year, dtype: float64

In [117]:
autos['registration_year'].describe()

count    46684.000000
mean      2002.910033
std          7.186122
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

75% of registration year is after 1999.
Old cars are rare on the list.

I investigate the relationship between brand and price below:

In [125]:
#Brand
autos['brand'].unique()

array(['peugeot', 'bmw', 'volkswagen', 'smart', 'ford', 'chrysler',
       'seat', 'renault', 'mercedes_benz', 'audi', 'sonstige_autos',
       'opel', 'mazda', 'porsche', 'mini', 'toyota', 'dacia', 'nissan',
       'jeep', 'saab', 'volvo', 'mitsubishi', 'jaguar', 'fiat', 'skoda',
       'subaru', 'kia', 'citroen', 'chevrolet', 'hyundai', 'honda',
       'daewoo', 'suzuki', 'trabant', 'land_rover', 'alfa_romeo', 'lada',
       'rover', 'daihatsu', 'lancia'], dtype=object)

In [149]:
autos['brand'].value_counts(normalize=True)

volkswagen        0.211293
bmw               0.110038
opel              0.107574
mercedes_benz     0.096457
audi              0.086561
ford              0.069917
renault           0.047147
peugeot           0.029839
fiat              0.025640
seat              0.018272
skoda             0.016408
nissan            0.015273
mazda             0.015187
smart             0.014159
citroen           0.014009
toyota            0.012702
hyundai           0.010025
sonstige_autos    0.009811
volvo             0.009147
mini              0.008761
mitsubishi        0.008226
honda             0.007840
kia               0.007069
alfa_romeo        0.006640
porsche           0.006126
suzuki            0.005934
chevrolet         0.005698
chrysler          0.003513
dacia             0.002635
daihatsu          0.002506
jeep              0.002271
subaru            0.002142
land_rover        0.002099
saab              0.001649
jaguar            0.001564
daewoo            0.001499
trabant           0.001392
r

I choose top 6 brand (percentage of the total values > 5%) to aggregate

In [168]:
brand_dict = {}

for b in range(0,6):
    price = autos[autos['brand'] == autos['brand'].value_counts().index[b]]['price'].mean()
    brand_dict[autos['brand'].value_counts().index[b]] = round(price)
    
brand_dict

{'volkswagen': 5604,
 'bmw': 8333,
 'opel': 2975,
 'mercedes_benz': 8628,
 'audi': 9337,
 'ford': 4055}

opel is cheapest. audi, bmv and mercedes are expensive.

In [167]:
mile_dict = {}

for b in range(0,6):
    mileage = autos[autos['brand'] == autos['brand'].value_counts().index[b]]['odometer_km'].mean()
    mile_dict[autos['brand'].value_counts().index[b]] = round(mileage)
    
mile_dict

{'volkswagen': 128711,
 'bmw': 132573,
 'opel': 129310,
 'mercedes_benz': 130788,
 'audi': 129157,
 'ford': 124266}

In [169]:
bmp_series = pd.Series(brand_dict)
bmm_series = pd.Series(mile_dict)

In [170]:
df = pd.DataFrame(bmp_series, columns=['mean_price'])
df['mean_mileage'] = bmm_series

In [171]:
df

mean_price  mean_mileage
volkswagen           5604        128711
bmw                  8333        132573
opel                 2975        129310
mercedes_benz        8628        130788
audi                 9337        129157
ford                 4055        124266

There is not siginigicant differece abount mean milage between each brand.  
Therefore, the differece about price can be regarded as the value of brand.
